war/src/test/java/io/lumeer/core/facade/CollectionFacadeIT.java

```diff
@@ -290,6 +289,42 @@ public void testUpdateCollectionAttributeAdd() {
       assertions.assertAll();
    }
 
+   @Test
+   public void testCreateCollectionAttribute() {
+      final Collection collection = createCollection(CODE);
+      assertThat(collection.getAttributes()).isEmpty();
+
+      final var attributeIds = new ArrayList<String>();
+
+      Runnable r = () -> {
+         for (int i = 0; i < 20; i++) {
+            Attribute attribute = new Attribute(null, ATTRIBUTE_NAME, null, null, 0);
+            final java.util.Collection<Attribute> createdAttributes = collectionFacade.createCollectionAttributes(collection.getId(), List.of(attribute));
+            createdAttributes.forEach(a -> attributeIds.add(a.getId()));
+         }
+      };
+
+      final List<Thread> threads = new ArrayList<>();
+      for (int i = 0; i < 4; i++) {
+         threads.add(new Thread(r));
+      }
+
+      threads.forEach(t -> t.run());
+
+      threads.forEach(t -> {
+         try {
+            t.join();
+         } catch (InterruptedException e) {
+            e.printStackTrace();
+         }
+      });
+
+
+      var updatedCollection = collectionDao.getCollectionByCode(CODE);
+      assertThat(updatedCollection).isNotNull();
+      assertThat(updatedCollection.getAttributes()).hasSize(4*20);
```

Comment:
Should not we test if all their IDs are unique? Because now we only check if all attributes get created.

Reply:
True, I wanted to have it stored in a set, not a list...